examples/machine_translation/train.py

```diff
@@ -0,0 +1,110 @@
+# Copyright 2022 The KerasNLP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import tensorflow as tf
+from absl import app
+from absl import flags
+
+from examples.machine_translation.data import get_dataset_and_tokenizer
+from examples.machine_translation.model import TranslationModel
+
+FLAGS = flags.FLAGS
+
+flags.DEFINE_integer("num_epochs", 1, "Number of epochs to train.")
+flags.DEFINE_integer("steps_per_epoch", None, "Number of steps per epoch.")
+flags.DEFINE_integer("num_encoders", 2, "Number of Transformer encoder layers.")
+flags.DEFINE_integer("num_decoders", 2, "Number of Transformer decoder layers.")
+flags.DEFINE_integer("batch_size", 64, "The training batch size.")
+flags.DEFINE_float("learning_rate", 0.001, "The initial learning rate.")
+flags.DEFINE_integer("embed_dim", 64, "Embedding size.")
```

Comment:
Should we rename this model_dim? model_hidden_dim? encoding_size?

I initially read this as the size of the embedding only, had to check rest of code to confirm this is also the main dimension for the encoder/decoder blocks.

Could alternately just leave a comment somewhere explaining.

Reply:
sg, I am renaming it to model_dim.